### PR TITLE
Adding a new CLI for consistent management of compression and encryption

### PR DESCRIPTION
**Pull request by lgriffin**
#### Please check if the PR fulfills these requirements

- [ ] The commit message is sensible and easily understood
- [ ] Tests for the changes have been added (for bug fixes / features where relevant)
- [ ] Docs have been added / updated (for bug fixes / features where relevant)

#### What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)

This is a feature addition to try and partly address Issue #689 wherein we need to have a consistent usage of encryption tooling and standards.  

#### Thank you for your contribution

We hope you stay around and connect with our growing community!

### CODE REVIEW

File: cli/tutors-json-manager/src/controller.ts [hunk 1]
```diff
@@ -0,0 +1,77 @@
+import { encryptData, decryptData } from './encryption';
+import { compressData, decompressData } from './compression';
+import logger from './logger';
+
+interface EncryptedCompressedData {
+  isCompressed: boolean;
+  isEncrypted: boolean;
+  data: string;
```
Comment (lgriffin) [inline cli/tutors-json-manager/src/controller.ts:8]: Structurally what I am aiming for here is two booleans to allow me know the order of operations as the encryption happens before the compression so the inverse has to occur in that order to successfully return the file

File: cli/tutors-json-manager/src/controller.ts [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+      isCompressed = true;
+      const compressedSize = Buffer.byteLength(jsonData, 'utf8');
+      const savedPercentage = ((originalSize - compressedSize) / originalSize) * 100;
+      logger.info(`Compression successful. Space saved: ${savedPercentage.toFixed(2)}%`);
+    }
+
+    return JSON.stringify({
+      isCompressed,
+      isEncrypted,
+      data: jsonData
```
Comment (lgriffin) [inline cli/tutors-json-manager/src/controller.ts:36]: This is the structure we are left with meaning we can quickly get a true/false read on the JSON that is created and the "payload" of jsonData is then free to interact with and parse later.

**Comment by lgriffin** [comment]
I added two single line comments to show one of the approaches that we can then reuse when this is applied elsewhere, it creates a simple boolean control included with the modified JSON payload. That should give us a simple if statement as a control, if the booleans are there, we decrypt and / or decompress, if they are not we have a plain text JSON.

**Comment by edeleastar** [comment]
Hi @lgriffin 

I have tried just focussing on the simple compress/decompress functions initially. I can successfully integrate into the generator, and have verified that they work successfully, achieving approx 5x compression.

However, when the compressed file is fetched in the reader, we get "incorrect header check" consistently. In debug, we can see the correct number of raw bytes coming down, pako cannot inflate correctly.

Perhaps it is related to the compression/encoding? Particularly compatibility with http/text?

In addition, the controller strategy looks interesting. It could, however, be shortcutted perhpas by the upgraded generator generating a "tutors.bin" instead of "tutors.json". The reader then can attempt to fetch the "bin" version. If present, assume it is encrypted and compressed. If not present, then we fall back to fetching the json?

**Pull request merged**